### PR TITLE
459 support site dont allow changing course from a fee paying to a salaried course

### PR DESCRIPTION
**Pull request by JonF1**
## Context
There was a recent issue where a candidate on a fee paying course was mistakenly moved to a salaried course in support

## Changes proposed in this pull request
* Flash warning message is shown if support agent tries to move candidate to a salaried course from a fee paying course telling the agent to raise a dev support ticket. 
* This occurs if the support agent tries to move the candidate when they are 'awaiting_provider_decision' and also after they have an offer and are in 'pending_conditions'.

## Guidance to review
1) Change a course choice to a salaried course. The candidate must be awaiting provider decision on a fee paying course.
2) Change an offered course to a salaried course. They must be in a pending conditions state on a fee paying course.

You should see the below message:

Screenshot 2022-08-10 at 15 05 42
Screenshot 2022-08-10 at 15 10 30

## Link to Trello card
https://trello.com/c/zuBT7efU/459-support-site-dont-allow-changing-course-from-a-fee-paying-to-a-salaried-course

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [ ] API release notes have been updated if necessary
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

**Comment by JonF1** [comment]
[Support site: don't allow changing course from a fee-paying to a salaried course](https://trello.com/c/zuBT7efU/459-support-site-dont-allow-changing-course-from-a-fee-paying-to-a-salaried-course)

File: app/models/course.rb [hunk 1]
```diff
@@ -123,6 +123,14 @@ def not_available?
...
   end
 
+  def fee_paying?
+    funding_type == 'fee'
+  end
+
+  def salaried?
+    funding_type == 'salary'
+  end
+
```
Comment (carlosmartinez) [inline app/models/course.rb:133]: do we also need to think about `apprenticeship`?

Comment (JonF1) [reply app/models/course.rb:133]: @MylesJarvis I assume that's ok as apprenticeship is also a type of salaried course?

Comment (MylesJarvis) [reply app/models/course.rb:133]: yep, I think best to cover all bases.

**Review by carlosmartinez — APPROVED** [review]

**Pull request merged**